src/Microsoft.Data/StringDataFrameColumn.cs

```diff
@@ -316,7 +316,11 @@ private StringDataFrameColumn CloneImplementation<U>(PrimitiveDataFrameColumn<U>
                         getBufferMaxRange = (bufferIndex + 1) * maxCapacity;
                     }
                     int bufferLocalMapIndex = (int)(mapIndex - getBufferMinRange);
-                    setBuffer[(int)index] = getBuffer[bufferLocalMapIndex];
+                    string value = getBuffer[bufferLocalMapIndex];
+                    setBuffer[(int)index] = value;
+                    if (value == null)
+                        ret._nullCount++;
```

Comment:
The existing unit tests didn't catch this `NullCount` mismatch because of a subtle difference in behavior in `Filter` vs `Sort`. Added a new unit test for this too now.